File: 169-majority-element/169-majority-element.java
class Solution 
{
    public int majorityElement(int[] nums) 
    {
        int countTarget = nums.length / 2;
        Random r = new Random();
        
        while(true)
        {
            int randIdx = r.nextInt(nums.length - 0) + 0;
            int randElement = nums[randIdx];
            int count = 0;
            for(int n : nums)
            {
                if(n == randElement)
                    count++;
                if(count > countTarget)
                    return randElement;
            }
            
            if(count > countTarget)
                return randElement;
        }
    }
}

File: 169-majority-element/169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        majority_freq = len(nums) // 2
        print(majority_freq)
        counts = {}
        for n in nums:
            curr_count = counts.get(n, 0) + 1
            if curr_count > majority_freq:
                return n
            counts[n] = curr_count